src/Search/Gearman/Command/QueueCommand.php

```diff
@@ -133,25 +135,44 @@ public function loop(InputInterface $input, LoggerInterface $logger)
         $logger->debug('-> Listening to topic ', ['topic' => $this->topic]);
         if ($this->queue->isValid()) {
             $item = $this->queue->dequeue($timeout);
-            // Transform into something for gearman.
-            $entity = $this->transformer->transform($item);
-            // Grab the gearman task.
-            $gearmanTask = $this->transformer->getGearmanTask($item);
-            // Run the task.
-            $logger->info('-> Running gearman task', [
-                'gearmanTask' => $gearmanTask,
-                'type' => $item->getType(),
-                'id' => $item->getId(),
-            ]);
-            // Set the task to go.
-            $this->client->doLow($gearmanTask, $entity, md5($item->getReceipt()));
-            // Commit.
-            $this->queue->commit($item);
-            $logger->info('-> Committed task', [
-                'gearmanTask' => $gearmanTask,
-                'type' => $item->getType(),
-                'id' => $item->getId(),
-            ]);
+            $entity = null;
+            try {
+                // Transform into something for gearman.
+                $entity = $this->transformer->transform($item);
+            } catch (BadResponse $e) {
+                $logger->error("Item does not exist in API: {$item->getType()} ({$item->getId()})", [
```

Comment:
It might be a blank string if SQS didn't return it, but it will be an instance of this: https://github.com/elifesciences/search/blob/develop/src/Search/Queue/QueueItem.php